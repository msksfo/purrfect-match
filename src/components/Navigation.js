import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/navigation.css';

const Navigation = () => {
    return (
        <nav className="navbar">
            <ul className="navbar--ul">
                <li className="navbar--li">
                    <Link to="/">Home</Link>
                </li>
                <li className="navbar--li">
                    <Link to="/about">About</Link>
                </li>
                <li className="navbar--li">
                    <Link to="/services">Services</Link>
                </li>
                <li className="navbar--li">
                    <Link to="/contact">Contact</Link>
                </li>
            </ul>
        </nav>
    );
};

export default Navigation;
