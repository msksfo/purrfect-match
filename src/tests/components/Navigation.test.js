import React from 'react';
import Navigation from '../../components/Navigation';
import { shallow } from 'enzyme';
import '../setupTests';

test('should correctly render the navigation component', () => {
    const wrapper = shallow(<Navigation />);
    expect(wrapper).toMatchSnapshot();
});
