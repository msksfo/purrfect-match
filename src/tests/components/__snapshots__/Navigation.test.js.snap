// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render the navigation component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="navbar--ul"
      >
        <li
          className="navbar--li"
        >
          Home
        </li>
        <li
          className="navbar--li"
        >
          About
        </li>
        <li
          className="navbar--li"
        >
          Services
        </li>
        <li
          className="navbar--li"
        >
          Contact
        </li>
      </ul>,
      "className": "navbar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="navbar--li"
          >
            Home
          </li>,
          <li
            className="navbar--li"
          >
            About
          </li>,
          <li
            className="navbar--li"
          >
            Services
          </li>,
          <li
            className="navbar--li"
          >
            Contact
          </li>,
        ],
        "className": "navbar--ul",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
            "className": "navbar--li",
          },
          "ref": null,
          "rendered": "Home",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "About",
            "className": "navbar--li",
          },
          "ref": null,
          "rendered": "About",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Services",
            "className": "navbar--li",
          },
          "ref": null,
          "rendered": "Services",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Contact",
            "className": "navbar--li",
          },
          "ref": null,
          "rendered": "Contact",
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="navbar--ul"
        >
          <li
            className="navbar--li"
          >
            Home
          </li>
          <li
            className="navbar--li"
          >
            About
          </li>
          <li
            className="navbar--li"
          >
            Services
          </li>
          <li
            className="navbar--li"
          >
            Contact
          </li>
        </ul>,
        "className": "navbar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="navbar--li"
            >
              Home
            </li>,
            <li
              className="navbar--li"
            >
              About
            </li>,
            <li
              className="navbar--li"
            >
              Services
            </li>,
            <li
              className="navbar--li"
            >
              Contact
            </li>,
          ],
          "className": "navbar--ul",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Home",
              "className": "navbar--li",
            },
            "ref": null,
            "rendered": "Home",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "About",
              "className": "navbar--li",
            },
            "ref": null,
            "rendered": "About",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Services",
              "className": "navbar--li",
            },
            "ref": null,
            "rendered": "Services",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Contact",
              "className": "navbar--li",
            },
            "ref": null,
            "rendered": "Contact",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
